UY 627 default cred

ppiernik opened the pull request.

ppiernik committed:
UY-627 add system credentials provider

ppiernik committed:
Merge remote-tracking branch 'origin/fr-620-out-of-the-box-setup' into UY-627-default-cred

ppiernik committed:
UY-627 add sys:all req, add repository for credential, fix unit test
configuration, first working version

ppiernik committed:
UY-627 add default configuration in distribution, small fixes

ppiernik committed:
UY-627 fix test-jar config, add posibility to set unchecked password

ppiernik committed:
UY-627 deep copy of system cred, add simple test

ppiernik committed:
Merge remote-tracking branch 'origin/fr-620-out-of-the-box-setup' into UY-627-default-cred

golbi commented on engine/src/main/java/pl/edu/icm/unity/engine/credential/CredentialManagementImpl.java:
```diff
@@ -179,4 +174,18 @@ private void checkEntityCredentialState(long entityId, LocalCredentialState desi
 					+ "the previous definition and can not keep the credential state as correct");
 	}
+	
+	private void assertIsNotSystemProfile(String name)
```
IsNotSystemCredential ?

golbi commented on engine/src/main/java/pl/edu/icm/unity/engine/credential/CredentialReqManagementImpl.java:
```diff
@@ -106,4 +112,16 @@ public void removeCredentialRequirement(String toRemove, String replacementId) t
 		credentialRequirementDB.delete(toRemove);
 	}
+	
+	private void assertIsNotSystemProfile(String name)
```
assertIsNotSystemCredReq

golbi commented on engine/src/main/java/pl/edu/icm/unity/engine/server/EngineInitialization.java:
```diff
@@ -555,17 +559,29 @@ private void initializeAdminUser()
 				PasswordToken ptoken = new PasswordToken(adminP);
-				idCredManagement.setEntityCredential(adminEntity, credDef.getName(), ptoken.toJson());
+			
+				//idCredManagement.setEntityCredential(adminEntity, credDef.getName(), ptoken.toJson());
```
remove

golbi commented on engine/src/main/resources/credentials/password.json:
```diff
@@ -0,0 +1,20 @@
+	"typeId": "password",
+	"name": "sys:password",
+	"readOnly": true,
+	"configuration": "{\"minLength\":10,\"historySize\":3,\"minClassesNum\":3,\"maxAge\":63058176000,\"denySequences\":true,\"allowLegacy\":false,\"scryptParams\":{\"workFactor\":17,\"blockSize\":8,\"parallelization\":1,\"length\":64},\"resetSettings\":{\"enable\":false}}",
```
after thinking - please disable mandatory password expiration - by default without this turned on

golbi commented on engine/src/main/java/pl/edu/icm/unity/engine/credential/EntityCredentialsHelper.java:
```diff
@@ -112,4 +111,62 @@ public void setPreviouslyPreparedEntityCredentialInternal(long entityId, String
+	 */
+	
+	
+	public void setEntityCredentialInternal(long entityId, String credentialId, 
```
please provide 2 methods, but without the boolean parameter -> more clarity

golbi reviewed (approved):
Very well, only some minor styling improvements. Also maybe a constant for the "sys:all" in code? And I'd add a new addEntity variant without credReq argument -> defaulting to sys:all. Then most of this sys:all usages scattered over many places can be reduced.

ppiernik committed:
UY-627 fix assert method name, support add entity without credReqId,
change "sys:passowrd" to constans DEFAULT_CREDENTIAL, remove max age of
default password
